examples/phi2/phi2_optimize_template.json

```diff
@@ -260,6 +258,17 @@
                     "source_max_len": 2048
                 }
             }
+        },
+        {
+            "name": "perf_tuning_data_config",
+            "type": "DummyDataContainer",
```

Comment:
this says dummydatacontainer but looks like this data config is supposed to only have a `dataloader_config`? What happens with the load_data component? Will it work even when the input shapes are missing?

Reply:
Not true. The `dataloader_config` is defaulted to `no_auto_batch_dataloader`.

Reply:
what I meant is this config doesn't appear to be using the dummy data container's load dataset component. it only needs the dataloader since it just ignores the dataset. 

Similar to https://github.com/microsoft/Olive/pull/1220#discussion_r1669373510, I don't know if the data config is even valid since it's missing input shapes. 

That's why I suggested https://github.com/microsoft/Olive/pull/1220#discussion_r1669361944

Reply:
Same here. Since the data config is used only for the purpose of perf tuning, the `input_shapes` come from `input_model/io_config`. However, the dataset that get created will be ignored since `create_phi2_dataloader` in phi2/user_script.py` ignores it and creates its own dataset/dataloder.

This is confusing and a roundabout way of implementing it, because phi2 doesn't actually implement an explicit dataset - it only has a dataloader. We could define an `EmpyDataContainer` but, in my opinion, that would be wrong path forward. Users should implement a dataset and not a dataloader. DataLoader is generic collecting logic and should be _smart_ - query the dataset batch size times and return the data - and so there shouldn't be a need to implement explicit DataLoaders.

Reply:
maybe in this example, it should explicitly bypass the load data component. or else, I think this would also fail without the input shapes https://github.com/microsoft/Olive/blob/7e6f5129a0d486c7a00cd27c62f57cd33dfe7922/olive/data/component/load_dataset.py#L38

Reply:
but like in the other discussion below, my concern is - if we are overriding all components, then using the type as dummydatacontainer or huggingfacecontainer is a bit confusing and/or misleading.

Reply:
I can see that as an argument but I don't think `EmptyContainer` is a solution. Needs further discussion and so will push it for a follow up PR.